### ISSUE TITLE
[deliver] Could not find a build for app: ... on platform: ios

### ISSUE THREAD

**Issue by ZwaanBanaan** [fastlane/fastlane#13696]
### Issue Description

fastlane crashes on when uploading an app to the appstore. The app is uploaded correctly and fastlane crashes afterwards:

``` 
/.gem-cache/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/ui/interface.rb:129:in `crash!': [!] Could not find a build for app: 1347102268 on platform: ios (FastlaneCore::Interface::FastlaneCrash)
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
2018-11-14T10:00:07.1598200Z [11:00:07]: Successfully uploaded the new binary to App Store Connect
2018-11-14T10:00:07.1611920Z [11:00:07]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
2018-11-14T10:00:08.6341860Z 
2018-11-14T10:00:08.6357730Z Looking for related GitHub issues on fastlane/fastlane...
2018-11-14T10:00:08.6364750Z 
2018-11-14T10:00:09.3527900Z Found no similar issues. To create a new issue, please visit:
2018-11-14T10:00:09.3542720Z https://github.com/fastlane/fastlane/issues/new
2018-11-14T10:00:09.3556200Z Run `fastlane env` to append the fastlane environment to your issue
2018-11-14T10:00:09.3573980Z [11:00:09]: WARNING: fastlane requires your locale to be set to UTF-8. To learn more go to https://docs.fastlane.tools/getting-started/ios/setup/#set-up-environment-variables
2018-11-14T10:00:09.3639210Z /Users/Company/.gem-cache/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/ui/interface.rb:129:in `crash!': [!] Could not find a build for app: 1347102268 on platform: ios (FastlaneCore::Interface::FastlaneCrash)
2018-11-14T10:00:09.3662840Z 	from /Users/Company/.gem-cache/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
2018-11-14T10:00:09.3687710Z 	from /Users/Company/.gem-cache/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/build_watcher.rb:13:in `wait_for_build_processing_to_be_complete'
2018-11-14T10:00:09.3711770Z 	from /Users/Company/.gem-cache/gems/fastlane-2.108.0/pilot/lib/pilot/build_manager.rb:55:in `upload'
2018-11-14T10:00:09.3732630Z 	from /Users/Company/.gem-cache/gems/fastlane-2.108.0/pilot/lib/pilot/commands_generator.rb:66:in `block (2 levels) in run'
2018-11-14T10:00:09.3753310Z 	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
2018-11-14T10:00:09.3773730Z 	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
2018-11-14T10:00:09.3806910Z 	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
2018-11-14T10:00:09.3831380Z 	from /Users/Company/.gem-cache/gems/fastlane-2.108.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
2018-11-14T10:00:09.3858190Z 	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
2018-11-14T10:00:09.3879530Z 	from /Users/Company/.gem-cache/gems/fastlane-2.108.0/pilot/lib/pilot/commands_generator.rb:166:in `run'
2018-11-14T10:00:09.3901340Z 	from /Users/Company/.gem-cache/gems/fastlane-2.108.0/pilot/lib/pilot/commands_generator.rb:18:in `start'
2018-11-14T10:00:09.3922870Z 	from /Users/Company/.gem-cache/gems/fastlane-2.108.0/fastlane/lib/fastlane/cli_tools_distributor.rb:100:in `take_off'
2018-11-14T10:00:09.3943960Z 	from /Users/Company/.gem-cache/gems/fastlane-2.108.0/bin/fastlane:23:in `<top (required)>'
2018-11-14T10:00:09.3963850Z 	from /usr/local/bin/fastlane:22:in `load'
2018-11-14T10:00:09.3983910Z 	from /usr/local/bin/fastlane:22:in `<main>'
2018-11-14T10:00:09.4098530Z ##[error]Error: fastlane failed with return code: 1
</pre>
</details>

### Environment

Fastlane-2.108.0

**Comment by janpio** [comment]
Are you using `deliver`/`upload_to_app_store` @ZwaanBanaan? 
Is `1347102268` the correct app id?
Did this happen only once or is this a regular occurence?

**Comment by cjmaroney01** [comment]
having this same issue, today.....just updated fastlane to most recent immediately before running and having this issue.

**Comment by ZwaanBanaan** [comment]
@janpio Sorry for the limited explination on this bug. I am making use of Visual Studio DevOps: 
https://marketplace.visualstudio.com/items?itemName=ms-vsclient.app-store

We have checked our app id's and they are all correct, the issue occures most of the time and it has always worked without any configuratios from ourselfs. Fastline automatically updates before each app release and in our app is pushed correctly in the appstore so luckely it has no direct problems for us, but it crashes afterwards.

**Comment by janpio** [comment]
Ok. 

I searched around a bit and found these similar issues: 
https://github.com/fastlane/fastlane/issues/13688
https://github.com/fastlane/fastlane/issues/13568

**Comment by lalitb** [comment]
+1 Same issue

**Comment by petkrein** [comment]
+1 same issue

**Comment by earlyster** [comment]
+1 same issue

**Comment by janpio** [comment]
Are you all using Visual Studio DevOps:
marketplace.visualstudio.com/items?itemName=ms-vsclient.app-store?

**Comment by Ajaxy** [comment]
Looks like it's a random error. Did not reproduce to me after restart.

**Comment by tmaly1980** [comment]
This happens for us every time. It's not a random error.

**Comment by otaro808** [comment]
In my case, I cannot use "wait_for_uploaded_build: true" because the terminal is hang out with this error:
"The request could not be completed because:
	Server error got 500"
So I use "skip_waiting_for_build_processing: true" as suggested on the terminal.

**Comment by awgeorge** [comment]
Could it be a race condition, reacting too quickly for apple to update their dashboard with the new build? I get this error randomly, but does occur almost always when sending 6+ apps to the store in parallelised processes. 

```
(FastlaneCore::Interface::FastlaneCrash)
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane_core/lib/fastlane_core/build_watcher.rb:13:in `wait_for_build_processing_to_be_complete'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/pilot/lib/pilot/build_manager.rb:55:in `upload'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
	from ../../../../../../../../../var/folders/b5/xq93wwkn5nzcbf24cf191p5c0000gn/T/fl_clone20190211-67584-gao454/fastlane.git/fastlane/Fastfile:121:in `block (2 levels) in parsing_binding'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/fastlane/lib/fastlane/cli_tools_distributor.rb:113:in `take_off'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.116.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

**Comment by awgeorge** [comment]
Actually - you know what - the problem is the build gets rejected by Apple. 

```
We identified one or more issues with a recent delivery for your app, "xxx". Please correct the following issues, then upload again.

Missing Purpose String in Info.plist File - Your app's code references one or more APIs that access sensitive user data. The app's Info.plist file should contain a NSBluetoothPeripheralUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. Starting Spring 2019, all apps submitted to the App Store that access user data will be required to include a purpose string.If you're using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. You can contact the developer of the library or SDK and request they release a version of their code that doesn't contain the APIs. Learn more (https://developer.apple.com/documentation/uikit/core_app/protecting_the_user_s_privacy).
```

So fast lane crashes as it's expecting the build to appear, but it can now apparently fail.

**Comment by awgeorge** [comment]
I corrected the build - apple approved and uploaded to TestFlight, however the same error persists. Interestingly on one of the three that failed, it looks like there is an error when a build had been rejected by apple.

Fastlane picks up the old rejected build, but that could be a different bug.
```
[09:13:31]: Successfully uploaded the new binary to App Store Connect
[09:13:31]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
[09:13:34]: Started watching build 2.2.0 - 302357 but expected 2.2.1 - 463209
[09:13:34]: Successfully finished processing the build 2.2.0 - 302357
[09:13:34]: Uploaded app 2.2.1 - 463209, but received build 2.2.0 - 302357. If you want to wait for uploaded build to be finished processing, use the `wait_for_uploaded_build` option
[09:13:36]: Successfully set the beta_app_feedback_email and/or beta_app_description
[09:13:39]: Successfully set the changelog for build
```
